lib/src/util/dart_type_utilities.dart

```diff
@@ -61,8 +61,8 @@ class DartTypeUtilities {
 
   /// Builds the list resulting from traversing the node in DFS and does not
   /// include the node itself.
-  static List<AstNode> traverseNodesInDFS(AstNode node) {
-    List<AstNode> nodes = [];
+  static LinkedHashSet<AstNode> traverseNodesInDFS(AstNode node) {
+    LinkedHashSet<AstNode> nodes = new LinkedHashSet();
```

Comment:
I don't understand this change. Please add a comment explaining why it's necessary that this be a set rather than a list.